packages/react-dom/src/events/ReactBrowserEventEmitter.js

```diff
@@ -117,31 +124,31 @@ export function listenTo(registrationName, contentDocumentHandle) {
 
   for (let i = 0; i < dependencies.length; i++) {
     const dependency = dependencies[i];
-    if (!(isListening.hasOwnProperty(dependency) && isListening[dependency])) {
-      if (dependency === 'topScroll') {
-        trapCapturedEvent('topScroll', 'scroll', mountAt);
-      } else if (dependency === 'topFocus' || dependency === 'topBlur') {
-        trapCapturedEvent('topFocus', 'focus', mountAt);
-        trapCapturedEvent('topBlur', 'blur', mountAt);
+    if (!isListening.get(dependency)) {
```

Comment:
There is a concern that using a Map could be slower.
I've seen this happen in https://github.com/facebook/react/pull/11733.

On the other hand, with numbers it would be slow to use number as a key in a regular object.

This needs benchmarking. Do you have ideas on how to measure the effect of this change?

Reply:
> Do you have ideas on how to measure the effect of this change?

Not really. One approach is probably to create a new benchmark example that fires random events - but that seems like a lot of work to get right. I'm also not sure how "realistic" that would be.

I tried to get an idea about this problem by running some micro benchmarks in Chrome, Firefox, Safari, IE11, and Edge: https://esbench.com/bench/5ade7596f2949800a0f61a8b

Seems like there's a huge variance across different browsers. The slow number lookup of a Map in Edge might be an issue.

![](https://dzwonsemrish7.cloudfront.net/items/3O0g3o351x090k1L3E2X/Screen%20Shot%202018-04-24%20at%2003.11.24.png?v=bf5643ec)